File: spring-boot-dto-angularjs/src/main/java/com/dicka/springbootdtoangularjs/service/UsersService.java
package com.dicka.springbootdtoangularjs.service;

import com.dicka.springbootdtoangularjs.dto.UsersDto;

import java.util.List;

public interface UsersService {

    UsersDto getUsersById(Integer usersId);
    void saveUsers(UsersDto usersDto);
    List<UsersDto> getAllUsers();
}

File: spring-boot-dto-angularjs/src/main/resources/static/js/usersService.js
'use strict'

angular.module('demo.service', []).factory('UsersService', ["$http", "CONSTANTS",
function($http, CONSTANTS){
    var service = {};
    service.getUsersById = function(usersId){
        var url = CONSTANTS.getUserByIdUrl + usersId;
        return $http.get(url);
    }
    service.getAllUsers = function(){
        return $http.get(CONSTANTS.getAllUsers);
    }
    service.saveUsers = function(usersDto){
        return $http.post(CONSTANTS.saveUsers, usersDto);
    }
    return service;
}]);